qa-pipelines/tasks/lib/klog-collection.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+__generate_klog() {
+  echo "Generating klog for namespace ${1}"
+  klog.sh "${1}"
+}
+
+upload_klogs_on_failure() {
+  # This function should run whenever the task exits with a failure.
+  # Task scripts should unset this as the EXIT handler before successful exits
+  local task_status=$?
+  echo "Task exited with status ${task_status}"
+  if [[ ${KLOG_COLLECTION_ON_FAILURE} != true ]]; then
+    echo "klog-collection-on-failure flag unset. Skipping container log aggregation"
+    return ${task_status}
+  fi
+  local klog_name=klog-$(date +%s)
+  set +o errexit
```

Comment:
Randomly unsetting `errexit` and not setting it back is fishy.

May need to check the output of `set -o` (with no arguments), or `$SHELLOPTS`.

Reply:
This is the last thing that runs in a task (`EXIT` is trapped with this function call), so errexit shouldn't be set. There may be situations like `helm delete scf` encountering a failure, where `klog.sh` may fail for cf, but we may still want klogs (and other information later) for `uaa`

Reply:
Well, yeah, I meant do we need to restore it at the end of the function.  Sorry, should have been clearer.

I guess nothing will (probably) run, though in the future we might want a different trap that calls this one manually instead.

Reply:
I actually did think about that initially (checking the errexit status at the beginning with `shopt -o errexit`, parsing the output for `on`/`off`, and setting it back to the pre-call value at the end) but it'd nearly double the length of this file for what seems like no benefit right now; if we need to add a cleanup handler later, the need for this will be immediately apparent on testing of that handler, so at that point we'd have to figure out a way to run both in a sensible order, and maybe would do something like that. Does that make sense?

Reply:
Actually, it might just be a few lines, I'll go ahead and add that in now